### DISCUSSION TITLE
Should I expose my Supabase URL and Anon Key to a public Github repo?

### DISCUSSION THREAD

**Discussion by marcoeribes** [supabase/supabase#22028] [Questions] [answered]
I have developed a react-app that is tightly coupled with supabase. That means if I don't have the supabase url or anon key in the react app, the contents of the app will not render. I published the react-app in a public repo and I've hidden url and key in a .env, but now when collaborators want to clone my repo, they cannot render the app because I have not pushed the .env file (it's in .gitignore). I'm afraid of publishing the anon key and url for potential security risks (although I'm not exactly sure what kind of risks those could be). Is there a better way of going about it, like developing a development supabase server? Utilizing local database? Any thoughts and suggestions would be appreciated!

**Comment by GaryAustin1** [comment, marked as answer]
I think you asked the same question on Discord.  I personally don't post the url/anon key in public places like github.  Although those values will be displayed in a browser when used in an app and are safe for security with RLS, there is no benefit to making it easy for scrapers to find. 

Normally telling a user to get a Supabase account and provide where to put the keys and what to do to set up the instance should be acceptable.  If you have a database just for demo purposes and don't intend to use it for your own app that might be an exception.
